### PR TITLE
MRG: Adding new variants to select the desired Qt binding

### LINKED ISSUE #12626: The PyQt6 dependency
Hi, we're currently depending on PyQt6, as there's a bug in PySide6 that has remained unaddressed for many years and breaks interactive Matplotlib use:

https://github.com/conda-forge/pyside2-feedstock/issues/226

Now, this wouldn't be much of a problem when using an all pip-installed environment. But on conda-forge, there is no PyQt6 package available (and I doubt there ever will be) – we only have PySide6 and PyQt5.

So when setting up my dev environment by first installing MNE from conda-forge, followed by a `pip install -e ".[dev,full]"` of `mne-python:main`, pip will start pulling in PyQt6 since it doesn't know that on my conda environment, I would rather use PyQt5 (or PySide6) instead.

So there's a discrepancy and this leads to a cluttered environment and segfaults.

(In case you're wondering why I'm not simply installing everything via pip from the get-go: I'm working on aarch64 (ARM Linux in a Docker container running on Apple Silicon), and many PyPI packages are not available as wheels for this platform).

I'm wondering what we could do about this. Several solutions (?) come to mind:

- Depend on PySide6 instead of PyQt and ask users to use IPython in case they depend on interactive plotting
- Add another installation variant that doesn't contain the PyQt dependencies, so I'm free to choose whichever Qt framework I want
- Add additional installation variants that explicitly depend either on PyQt5 or PySide6

I'd love to hear your feedback. This is truly a very frustrating situation for me.

Thanks!

cc @cbrnr

### PR DESCRIPTION
**Pull request by hoechenberger**
This adds:

- `mne[full-no-qt]`: no Qt bindings
- `mne[full-pyqt6]`: equivalent to `mne[full]`, Qt6 bindings
- `mne[full-pyside6]`: PySide6 bindings

Fixes #12626

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -59,16 +59,18 @@ data = []
...
     "mne[hdf5]",
     "qtpy",
-    "PyQt6!=6.6.0",
-    "PyQt6-Qt6!=6.6.0,!=6.7.0",
     "pyobjc-framework-Cocoa>=5.2.0; platform_system=='Darwin'",
     "sip",
     "scikit-learn",
     "nibabel",
-    "openmeeg>=2.5.5",
+    # "openmeeg>=2.5.5",
```
Comment (hoechenberger) [inline pyproject.toml:73]:
```suggestion
    "openmeeg>=2.5.5",
```

File: pyproject.toml [hunk 2]
```diff
@@ -106,6 +108,19 @@ full = [
...
+    "PyQt6!=6.6.0",
+    "PyQt6-Qt6!=6.6.0,!=6.7.0",
+]
+full-pyqt6 = [
+    "mne[full]"
+]
+full-pyside6 = [
+    "mne[full-no-qt]",
+    "PySide6"
+]
```
Comment (hoechenberger) [inline pyproject.toml:122]: Actually … we could also do:
```python
pyqt6 = []
pyside6 = [
    "PySide6"
]
```
and then the user could install it by selecting `mne[full-no-qt,pyside6]` … or would that be awkward?

**Comment by hoechenberger** [comment]
💡 this would also allow us to to more easily test with different bindings

**Review by mscheltienne — APPROVED** [review]
LGTM, I like the 3 options `full`, `full-pyqt6` and `full-pyside6`.

**Comment by mscheltienne** [comment]
Doc build will be fixed by #12625

**Review by larsoner — APPROVED** [review]

**Comment by hoechenberger** [comment]
@larsoner I've disabled auto-merge, I need to push some formatting fixed and also a changelog entry

**Comment by drammock** [comment]
> @larsoner I've disabled auto-merge, I need to push some formatting fixed and also a changelog entry



> @larsoner I've disabled auto-merge, I need to push some formatting fixed and also a changelog entry

... and also a documentation update describing this change, please

**Comment by hoechenberger** [comment]
@drammock I think I've addressed your comments

**Review by drammock — APPROVED** [review]

**Pull request merged**